### PR TITLE
Improve ViT device and e2e time

### PR DESCRIPTION
**Pull request by pavlepopovic**
### Ticket
[Link to Github Issue
](https://github.com/tenstorrent/tt-metal/issues/23082)

### Problem description
ViT is slow on main

### What's changed
Speed it up by doing the following:
- entire model except attention matmul->softmax->matmul part to run on 64 cores instead of 48
- add reshards in between
- remove attention mask
- optimise layernorm
- remove unnecessary reallocates

Now, on N150 device perf is ~1430 samples/s, and e2e is ~1377.
On N300 one chip, device perf is the same, but E2E is lower and in the range of ~1250. I'm guessing this is due to ETH dispatch being slow, but have no solid proof at the minute. To be investigated.

In addition to speeding it up, add device & e2e perf tests to ci.

### Checklist
- [x] [All post commit][(https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes](https://github.com/tenstorrent/tt-metal/actions/runs/15453798672)
- [x] [Model regression] https://github.com/tenstorrent/tt-metal/actions/runs/15467264111/job/43542684652 falcon7b huggingface failure, newly added test is passing
- [x] [Device performance regression] https://github.com/tenstorrent/tt-metal/actions/runs/15453772901
- [x]  [Frequent model tests] https://github.com/tenstorrent/tt-metal/actions/runs/15453779107 unrelated whisper failure that seems to come from a machine issue?

### CODE REVIEW

**Review by mbahnasTT — APPROVED** [review]

File: models/demos/wormhole/vit/demo/test_vit_device_perf.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+@pytest.mark.parametrize(
+    "expected_kernel_samples_per_sec",
+    [
+        1450,
+    ],
+)
+@pytest.mark.models_device_performance_bare_metal
+def test_vit_perf_device(batch_size, expected_kernel_samples_per_sec):
+    command = f"pytest models/demos/wormhole/vit/demo/test_vit_device_perf.py::test_vit_device_ops[{batch_size}-device_params0]"
+    print("Generated command is: ", command)
```
Comment (bbradelTT) [inline models/demos/wormhole/vit/demo/test_vit_device_perf.py]: Could logger be used instead of print?

**Review by bbradelTT — APPROVED** [review]

File: models/demos/wormhole/vit/demo/test_vit_device_perf.py (hunk 1 above)
Comment (pavlepopovic) [reply models/demos/wormhole/vit/demo/test_vit_device_perf.py]: Thx for catching this, don't really need this print either in log() or print() format, so I'm gonna remove it :)

**Review by kkabilarTT — APPROVED** [review]
